x-pack/plugins/fleet/server/services/fleet_server_migration.ts

```diff
@@ -8,10 +8,45 @@ import { KibanaRequest } from 'src/core/server';
 import {
   ENROLLMENT_API_KEYS_INDEX,
   ENROLLMENT_API_KEYS_SAVED_OBJECT_TYPE,
+  AGENT_POLICY_INDEX,
+  AGENT_ACTIONS_INDEX,
+  AGENTS_INDEX,
   FleetServerEnrollmentAPIKey,
+  FleetServerPackage,
 } from '../../common';
 import { listEnrollmentApiKeys, getEnrollmentAPIKey } from './api_keys/enrollment_api_key_so';
 import { appContextService } from './app_context';
+import { getInstallation } from './epm/packages';
+
+export async function isFleetServerSetup() {
+  const pkgInstall = await getInstallation({
+    savedObjectsClient: getInternalUserSOClient(),
+    pkgName: FleetServerPackage,
+  });
+
+  if (!pkgInstall) {
+    return false;
+  }
+
+  const esClient = appContextService.getInternalUserESClient();
+  const indexes = [
```

Comment:
the code intention is different (just checking for existence here, rather than checking + creating if indices don't exist), so I'm not sure how much de-dupe is valuable. it would be nice to at least export another constant that is an array of all Fleet server indices like `FLEET_SERVER_INDICES` to use here and in `ensureFleetServerIndicesCreated`